### PR TITLE
Add KZGProof at `G1_POINT_AT_INFINITY` test cases

### PR DESCRIPTION
**Pull request by hwwhww**

### CODE REVIEW

File: tests/generators/kzg_4844/main.py [hunk 1]
```diff
@@ -399,6 +399,20 @@ def case05_verify_blob_kzg_proof():
             'output': None
         }
 
+    # Edge case: Incorrect `G1_POINT_AT_INFINITY` Proof
+    blob = BLOB_RANDOM_VALID1
+    commitment = spec.blob_to_kzg_commitment(blob)
+    proof = spec.G1_POINT_AT_INFINITY
+    assert not spec.verify_blob_kzg_proof(blob, commitment, proof)
+    yield 'verify_blob_kzg_proof_case_incorrect_proof_point_at_infinity', {
```
Comment (jtraglia) [inline tests/generators/kzg_4844/main.py:407]: Minor nit. I think we should group this with the other "incorrect" tests. So move this test case below the `verify_blob_kzg_proof_case_incorrect_proof` test case.

File: tests/generators/kzg_4844/main.py [hunk 2]
```diff
@@ -399,6 +399,20 @@ def case05_verify_blob_kzg_proof():
             'output': None
         }
 
+    # Edge case: Incorrect `G1_POINT_AT_INFINITY` Proof
```
Comment (jtraglia) [inline tests/generators/kzg_4844/main.py:402]: For consistency:
```suggestion
    # Incorrect `G1_POINT_AT_INFINITY` proof
```

File: tests/generators/kzg_4844/main.py [hunk 3]
```diff
@@ -509,6 +523,21 @@ def case06_verify_blob_kzg_proof_batch():
         'output': None
     }
 
+    # Edge case: Incorrect `G1_POINT_AT_INFINITY` Proof
```
Comment (jtraglia) [inline tests/generators/kzg_4844/main.py:526]: For consistency:
```suggestion
    # Incorrect `G1_POINT_AT_INFINITY` proof
```

File: tests/generators/kzg_4844/main.py [hunk 4]
```diff
@@ -509,6 +523,21 @@ def case06_verify_blob_kzg_proof_batch():
...
+    blob = BLOB_RANDOM_VALID1
+    commitment = spec.blob_to_kzg_commitment(blob)
+    proof = spec.G1_POINT_AT_INFINITY
+    assert not spec.verify_blob_kzg_proof_batch([blob], [commitment], [proof])
+
+    yield 'verify_blob_kzg_proof_batch_case_incorrect_proof_point_at_infinity', {
```
Comment (jtraglia) [inline tests/generators/kzg_4844/main.py:532]: Nit: I think it looks more consistent without the blank line:
```suggestion
    assert not spec.verify_blob_kzg_proof_batch([blob], [commitment], [proof])
    yield 'verify_blob_kzg_proof_batch_case_incorrect_proof_point_at_infinity', {
```

Comment (jtraglia) [inline tests/generators/kzg_4844/main.py:532]: Like the other one, I think this should be in the "incorrect" group of tests. So below the `verify_blob_kzg_proof_batch_case_incorrect_proof_add_one` test.

**Review by jtraglia — CHANGES_REQUESTED** [review]
Good idea for this test case! Just some small nits.

**Comment by jtraglia** [comment]
Also, I can confirm that these tests pass in c-kzg-4844.

**Comment by mratsim** [comment]
Cc @etan-status who mentioned an issue related to this

**Pull request merged**